guides/06_client-libraries/02_getting-started-with-the-js-client.md

```diff
@@ -0,0 +1,251 @@
+
+## Connecting to a Hugging Face Space
+
+```js
+import { client } from "@gradio/client";
+
+const app = client("abidlabs/en2fr"); // a Space that translates from English to French
+```
+
+You can also connect to private Spaces by passing in your HF token with the `hf_token` property of the options parameter. You can get your HF token here: https://huggingface.co/settings/tokens
+
+```js
+import { client } from "@gradio/client";
+
+const app = client("abidlabs/my-private-space", { hf_token="hf_..." }) 
+```
+
+
+## Duplicating a Space for private use
+
+While you can use any public Space as an API, you may get rate limited by Hugging Face if you make too many requests. For unlimited usage of a Space, simply duplicate the Space to create a private Space, and then use it to make as many requests as you'd like! 
+
+The `@gradio/client` exports another function, `duplicate`, to make this process simple (you'll need to pass in your [Hugging Face token](https://huggingface.co/settings/tokens)). 
+
+`duplicate`` is almost identical to `client`, the only difference is under the hood:
+
+```js
+import { client } from "@gradio/client";
+
+const response = await fetch(
+  "https://audio-samples.github.io/samples/mp3/blizzard_unconditional/sample-0.mp3"
+);
+const audio_file = await response.blob();
+
+const app = client("abidlabs/whisper", { hf_token="hf_..." });
+const transcription = app.predict("/predict", [audio_file]);
+```
+
+If you have previously duplicated a Space, re-running `duplicate` will *not* create a new Space. Instead, the client will attach to the previously-created Space. So it is safe to re-run the `duplicate` method multiple times with the same space. 
+
+**Note:** if the original Space uses GPUs, your private Space will as well, and your Hugging Face account will get billed based on the price of the GPU. To minimize charges, your Space will automatically go to sleep after 1 hour of inactivity. You can also set the hardware using the `hardware` parameter of `duplicate()`.
+
+
+## Connecting a general Gradio app
+
+If your app is running somewhere else, just provide the full URL instead, including the "http://" or "https://". Here's an example of making predictions to a Gradio app that is running on a share URL:
+
+```js
+import { client } from "@gradio/client";
+
+const app = client("https://bec81a83-5b5c-471e.gradio.live");
+```
+
+## Inspecting the API endpoints
+
+Once you have connected to a Gradio app, you can view the APIs that are available to you by calling the `client`'s `view_api` method. For the Whisper Space, we see the following:
+
+```json
+{
+  "named_endpoints": {
+    "/predict": {
+      "parameters": [
+        {
+          "label": "text",
+          "component": "Textbox",
+          "type": "string"
+        }
+      ],
+      "returns": [
+        {
+          "label": "output",
+          "component": "Textbox",
+          "type": "string"
+        }
+      ]
+    }
+  },
+  "unnamed_endpoints": {}
+}
```

Comment:
How can I see the full json output? When I ran

```
console.log(app.view_api());
```
All of the inner dictionaries were just printed as [Object object]. 

When I ran
```js
console.log(JSON.stringify(app.view_api()));
```

an empty dictionary was printed

Reply:
This also returns a Promise. This should work:

```js
console.log(JSON.stringify(await app.view_api()))
```

Or for more clarity:

```js
const endpoint_info = await app.view_api();
console.log(JSON.stringify(endpoint_info))
```

Reply:
Hmm now I'm getting:

`TypeError: app.view_info is not a function`